lib/src/working_copy.rs

```diff
@@ -516,18 +516,35 @@ impl TreeState {
                 source: err,
             }
         })?;
-        self.tree_id = TreeId::new(proto.tree_id.clone());
+        #[allow(deprecated)]
+        if !proto.legacy_tree_id.is_empty() {
```

Comment:
nit: I slightly prefer testing `proto.tree_ids.is_empty()` since `legacy_tree_id.is_empty()` checks length of the id.

Reply:
Done